tss-esapi/src/structures/buffers/public.rs

```diff
@@ -0,0 +1,520 @@
+                        error!("ECC parameters have not ben set in the Public Builder even though the ECC algorithm have been selected");
+                        Error::local_error(WrapperErrorKind::ParamsMissing)
+                    })?,
+                    unique: self.ecc_unique_identifier.ok_or_else(|| {
+                        error!("ECC unique identifier have not ben set in the Public Builder even though the ECC algorithm have been selected");
+                        Error::local_error(WrapperErrorKind::ParamsMissing)
+                    })?,
+                })
+            }
+            PublicAlgorithm::SymCipher => {
+                Ok(Public::SymCipher {
+                    object_attributes,
+                    name_hashing_algorithm,
+                    auth_policy,
+                    parameters: self.symmetric_cipher_parameters.ok_or_else(|| {
+                        error!("Symmetric cipher parameters have not ben set in the Public Builder even though the symmetric cipher algorithm have been selected");
+                        Error::local_error(WrapperErrorKind::ParamsMissing)
+                    })?,
+                    unique: self.symmetric_cipher_unique_identifier.ok_or_else(|| {
+                        error!("Symmetric cipher unique identifier have not ben set in the Public Builder even though the symmetric cipher algorithm have been selected");
+                        Error::local_error(WrapperErrorKind::ParamsMissing)
+                    })?,
+                })
+            }
+        }
+    }
+}
+
+/// Enum representing the Public structure.
+///
+/// # Details
+/// This corresponds to TPM2B_PUBLIC
+#[derive(Debug, Clone)]
+pub enum Public {
+    Rsa {
+        object_attributes: ObjectAttributes,
+        name_hashing_algorithm: HashingAlgorithm,
+        auth_policy: Digest,
+        parameters: PublicRsaParameters,
+        unique: PublicKeyRsa,
+    },
+    KeyedHash {
+        object_attributes: ObjectAttributes,
+        name_hashing_algorithm: HashingAlgorithm,
+        auth_policy: Digest,
+        parameters: PublicKeyedHashParameters,
+        unique: Digest,
+    },
+    Ecc {
+        object_attributes: ObjectAttributes,
+        name_hashing_algorithm: HashingAlgorithm,
+        auth_policy: Digest,
+        parameters: PublicEccParameters,
+        unique: EccPoint,
+    },
+    SymCipher {
+        object_attributes: ObjectAttributes,
+        name_hashing_algorithm: HashingAlgorithm,
+        auth_policy: Digest,
+        parameters: SymmetricCipherParameters,
+        unique: Digest,
+    },
+}
+
+impl Public {
+    /// Returns the object attributes
+    pub const fn object_attributes(&self) -> ObjectAttributes {
+        match self {
+            Public::Rsa {
+                object_attributes,
+                name_hashing_algorithm: _,
+                auth_policy: _,
+                parameters: _,
+                unique: _,
+            } => *object_attributes,
```

Comment:
```suggestion
            Public::Rsa { object_attributes, .. } => *object_attributes,
```
At least I hope that's the syntax

Reply:
Thank you I was looking for that but I couldn't find it.

Reply:
I have fixed so they look like this now. Very nice.